### ISSUE TITLE
Normalize OS detection

### ISSUE THREAD

**Issue by pdehaan** [rpappalax/ff-tool#20]
Currently `get_os()` returns `platform.system().lower()` (and then sprinkles on some magic). This is "funny" because on Windows+Cygwin, it could return "windows" or "cygwin", meaning it will try and load configs/windows.ini (on our Win10 laptop), or configs/cygwin.ini (on our Win2012 server). I was seeing similar jank before for Linux as well.

So we should probably tweak the `is_windows()` to determine if it is running Windows or Cygwin and then `get_os()` will always return "windows".

```py
    @staticmethod
    def get_os():
        """
        Do our best to determine the user's current operating system.
        """
        system = platform.system().lower()
        return re.split('[-_]', system, maxsplit=1).pop(0)
```

**Comment by pdehaan** [comment]
I think the other issue I was seeing was the OS reporting as either "linux" or "linux-gnu" depending on whether I was trying to run scripts on AWS versus Docker locally, but IIRC that was sorted by doing the `maxsplit=1` and `.pop(0)` tricks above.

**Comment by pdehaan** [comment]
We added this in the `cygwin` branch (which has been merged into `dev` as per PR #21).
Closing.
